### ISSUE TITLE
Draw the image data in the runs folder into an editable window in MATLAB

### ISSUE THREAD

**Issue by listening510** [matlab-deep-learning/Pretrained-YOLOv8-Network-For-Object-Detection#27]
After training YOLOv8 in MATLAB and obtaining yolov8Det, I’m wondering if there’s a code snippet or method to extract the data of automatically generated images—such as ROC curves and confusion matrices—from the "runs" folder created by the Ultralytics library. I want to save this data to the workspace, and once the training code finishes running, generate corresponding MATLAB figure windows for further editing.
I’m using MATLAB 2024b, Python 3.11, and have downloaded the Ultralytics library as you recommended. Thank you very much for your help and answers.
I noticed that the results.csv file in the "runs" folder does not include data related to ROC curves, and the automatically generated ROC images in the folder do not meet the requirements for scientific plotting standards.Thank you for your answer

**Comment by viakkala** [comment]
Thanks for raising this request it’s a very valid need, especially for users who require scientific‑quality plots and traceable numerical data for publications or reporting. We’ll treat this as an enhancement request.

Currently, MATLAB’s YOLOv8 training workflow integrates with Python mainly for model training and inference, but it does not provide built‑in support for parsing the Ultralytics runs folder or extracting internal metrics such as ROC curve points or confusion‑matrix values directly into the MATLAB workspace.

Ultralytics generates ROC curves and confusion matrices inside Python during training and saves them only as image files. MATLAB does not currently expose an API to retrieve the underlying data used to create those plots.

### Workaround: Recompute Metrics in MATLAB
A practical solution for producing publication‑quality MATLAB plots is to recompute these evaluation metrics using MATLAB’s built‑in functions.

Refer to the [Evaluate Detector Using Test Set section](https://www.mathworks.com/help/vision/ug/object-detection-using-yolov4-deep-learning.html#ObjectDetectionUsingYOLOV4DeepLearningExample-6) in the doc for more details on evalaution:

Below is an example snippet:

detectionResults = detect(detector, testData, Threshold=0.01);
metrics = evaluateObjectDetection(detectionResults, testData);

AP = averagePrecision(metrics);
[precision, recall] = precisionRecall(metrics, ClassName="vehicle");

figure
plot(recall{:}, precision{:});
xlabel("Recall");
ylabel("Precision");
grid on;
title(sprintf("Average Precision = %.2f", AP));
